src/Domain/Infra/InMemory/DomainEntityRepositoryTrait.php

```diff
@@ -28,58 +30,54 @@ private function doFindAll(int $offset = 0, int $limit = 0): DomainCollectionInt
 
     private function doFindAllByFields(array $fields, int $offset = 0, int $limit = 0): DomainCollectionInterface
     {
+        if (!$fields) {
+            throw new \LogicException('No fields provided.');
+        }
+
+        $i = -1;
         $entities = [];
         foreach ($this->memory->all($this->class) as $entity) {
-            // @todo duplicated in doFindByFields
-            foreach ($fields as $field => $value) {
-                $knownValue = $this->getEntityField($entity, $field);
-                if ($knownValue instanceof DomainIdInterface && $value instanceof DomainIdInterface && $knownValue->equals($value)) {
-                    continue;
-                }
-
-                if ($value !== $knownValue) {
-                    continue 2;
-                }
+            if (!$this->matchesFields($entity, $fields) || ++$i < $offset) {
+                continue;
+            } elseif ($limit && $i >= ($offset + $limit)) {
+                break;
             }
 
             $entities[] = $entity;
         }
 
-        return $this->createResultSet($entities, $offset, $limit);
+        return $this->createResultSet($entities);
     }
 
     private function doFind($id, ...$idN)
     {
-        return $this->doFindByFields(array_combine($this->idFields, func_get_args()));
+        if (!$this->isValidEntityId($ids = func_get_args())) {
```

Comment:
`doFind(...$idN)` would solve the problem I think

Reply:
idea is we disallow `doFind()` by design, api requires at least one known id value. Same goes for the above really. It's truly a different signature.

Reply:
perhaps for the trait it might be more pracitcal; as the signature is enforced by the wrapper anyway, e.g. `UserRepo::find`. But in theory you could use the trait to create a generic entity repository, i.e. `use RepoTrait { doFind as public find; }`.